File: javascript/countgoodnodesinbinarytree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var goodNodes = function(root) {
    let good=0;
    function dfs(node,path){
        if (node===null){
            
            node.val>=Math.max(path) ? (good++) : (good);
            path.push(node.val);
            dfs(node.left,path);
            dfs(node.right,path);
            path.pop();
            return path;
        }
    }
    dfs(root,[]);
    return good;
    
};

File: python/countgoodnodesinbinarytree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def __init__(self) -> None:
        self.good=0
    def goodNodes(self, root: TreeNode) -> int:
        def dfs(node,path):
            if node:
                if len(path)==0 or node.val>=max(path):
                    self.good+=1
                path.append(node.val)
                print(path)
                left_path=path
                right_path=path
                dfs(node.left,left_path)
                dfs(node.right,right_path)
                path.pop()
            return self.good
        
        return dfs(root,[])
        



                
                
        